src/redisearch_rs/rlookup/src/lookup.rs

```diff
@@ -239,10 +242,62 @@ impl<'a> RLookupKey<'a> {
         // Safety: 3 -> Caller has to uphold the pin contract
         unsafe { Pin::new_unchecked(b) }
     }
+
+    /// Updates this `RLookupKey`'s fields according to the provided [`ffi::FieldSpec`].
+    pub fn update_from_field_spec(&mut self, fs: &ffi::FieldSpec) {
+        self.flags |= RLookupKeyFlag::DocSrc | RLookupKeyFlag::SchemaSrc;
+
+        let path = {
+            debug_assert!(!fs.fieldPath.is_null());
+
+            let mut path_len = 0;
+            // Safety: we received the pointer from the field spec and have to assume it is valid
+            let path_ptr =
+                unsafe { ffi::HiddenString_GetUnsafe(fs.fieldPath, ptr::from_mut(&mut path_len)) };
+
+            debug_assert!(!path_ptr.is_null());
+            // Safety: We assume the `path_ptr` and `length` information returned by the field spec
+            // point to a valid null-terminated C string. Importantly `length` here is value as returned by
+            // `strlen` so **does not** include the null terminator (that is why we do `path_len + 1` below)
+            let bytes = unsafe { slice::from_raw_parts(path_ptr.cast::<u8>(), path_len + 1) };
+            let path = CStr::from_bytes_with_nul(bytes)
+                .expect("string returned by HiddenString_GetUnsafe is malformed");
+
+            // When the name is owned, we also want the path to be owned
+            if matches!(self._name, Cow::Owned(_)) {
+                Cow::Owned(path.to_owned())
+            } else {
+                Cow::Borrowed(path)
+            }
+        };
+        self._path = Some(path);
+        self.path = self._path.as_ref().unwrap().as_ptr();
+
+        let fs_options = FieldSpecOptions::from_bits(fs.options()).unwrap();
+
+        if fs_options.contains(FieldSpecOption::Sortable) {
+            self.flags |= RLookupKeyFlag::SvSrc;
+            self.svidx = u16::try_from(fs.sortIdx).unwrap();
```

Comment:
Similar question here - can we rely on the values coming from C and be more relaxed when reading them, e.g., use `from_unchecked`? If not, can add a clear error message using `expect`?

Reply:
yes we can, see above.